gn3/computations/principal_component_analysis.py

```diff
@@ -0,0 +1,182 @@
+"""module contains pca implementation using python"""
+
+
+from typing import Any
+from scipy import stats
+
+from sklearn.decomposition import PCA
+from sklearn import preprocessing
+
+import numpy as np
+
+
+from typing_extensions import TypeAlias
+
+fArray: TypeAlias = list[float]
+
+
+def compute_pca(array: list[fArray]) -> dict[str, Any]:
+    """
+    computes the principal component analysis
+
+    Parameters:
+
+          array(list[list]):a list of lists contains data to perform  pca
+
+
+    Returns:
+           pca_dict(dict):dict contains the pca_object,pca components,pca scores
+
+
+    """
+
+    corr_matrix = np.matrix(array)
+
+    pca_obj = PCA()
+    scaled_data = preprocessing.scale(corr_matrix)
+
+    pca_obj.fit(scaled_data)
+
+    return {
+        "pca": pca_obj,
+        "components": pca_obj.components_,
+        "scores": pca_obj.transform(scaled_data)
+    }
+
+
+def generate_scree_plot_data(variance_ratio: fArray) -> list[tuple[str, float]]:
+    """
+    generates the scree data for plotting
+
+    Parameters:
+
+            variance_ratio(list[floats]):ratios for contribution of each pca
+
+    Returns:
+
+            coordinates(list[(x_coor,y_coord)])
+
+
+    """
+
+    perc_var = [round(ratio*100, 1) for ratio in variance_ratio]
+
+    # perc_var = np.round(variance_ratio*100, decimals=1)
+
+    x_coordinates = [f"PC{val}" for val in range(1, len(perc_var)+1)]
+
+    return list(zip(x_coordinates, perc_var))
```

Comment:
**issue**: Why not just return the generator and let the caller do the casting, or looping?

Reply:
What exactly do you mean here? Would this just mean replacing `return list(zip(x_coordinates, perc_var))` with `yield zip(x_coordinates, per_var)` and calling the function like `[result for result in generate_scree_plot_data(variance_radio)]`?

Reply:
made modification to this rather than returning each pair of coordinates I return the both  lists and leave  the rest for plotly
to handle